3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/SendgridAdapter.scala

```diff
@@ -69,53 +51,72 @@ object SendgridAdapter extends Adapter {
   private val TrackerVersion = "com.sendgrid-v3"
 
   // Schemas for reverse-engineering a Snowplow unstructured event
-  private val EventSchemaMap = Map (
-    "subscribe" -> SchemaKey("com.sendgrid", "new", "jsonschema", "1-0-0").toSchemaUri
+  private val EventSchemaMap = Map(
+    "processed" -> SchemaKey("com.sendgrid", "processed", "jsonschema", "1-0-0").toSchemaUri,
+    "dropped" -> SchemaKey("com.sendgrid", "dropped", "jsonschema", "1-0-0").toSchemaUri,
+    "delivered" -> SchemaKey("com.sendgrid", "delivered", "jsonschema", "1-0-0").toSchemaUri,
+    "deferred" -> SchemaKey("com.sendgrid", "deferred", "jsonschema", "1-0-0").toSchemaUri,
+    "bounce" -> SchemaKey("com.sendgrid", "bounce", "jsonschema", "1-0-0").toSchemaUri,
+    "open" -> SchemaKey("com.sendgrid", "open", "jsonschema", "1-0-0").toSchemaUri,
+    "click" -> SchemaKey("com.sendgrid", "click", "jsonschema", "1-0-0").toSchemaUri,
+    "spamreport" -> SchemaKey("com.sendgrid", "spamreport", "jsonschema", "1-0-0").toSchemaUri,
+    "unsubscribe" -> SchemaKey("com.sendgrid", "unsubscribe", "jsonschema", "1-0-0").toSchemaUri,
+    "group_unsubscribe" -> SchemaKey("com.sendgrid", "group_unsubscribe", "jsonschema", "1-0-0").toSchemaUri,
+    "group_resubscribe" -> SchemaKey("com.sendgrid", "group_resubscribe", "jsonschema", "1-0-0").toSchemaUri
   )
 
+
+  def payloadBodyToEvents(body: String, payload: CollectorPayload): List[Validated[RawEvent]] = {
+
+    parse(body).children.map(itm => {
+
+      val eventType = (itm \\ "event").extract[String]
```

Comment:
If the JSON you parsed does not find event I believe this will throw an exception.  To be safer you can do this:

`val eventType: Option[String] = (itm \\ "event").extractOpt[String]`

This removes the need to use eventType.some in your match statement as well.